src/s3/util/fileViewerManager.ts

```diff
@@ -65,13 +113,32 @@ export class S3FileViewerManager {
         }
         getLogger().verbose(`S3FileViewer: File from s3 or temp to be opened is: ${fileLocation}`)
 
-        const newTab = this.activeTabs.get(fileLocation.fsPath) ?? new S3Tab(fileLocation)
+        const newTab = this.activeTabs.get(fileLocation.fsPath) ?? new S3Tab(fileLocation, fileNode)
         await newTab.openFileInReadOnly()
 
         this.activeTabs.set(fileLocation.fsPath, newTab)
     }
 
     public async openInEditMode(uriOrNode: vscode.Uri | S3FileNode): Promise<void> {
+        if (this.promptOnEdit) {
+            const message = localize(
+                'AWS.s3.fileViewer.warning.editStateWarning',
+                'Opening S3 file for editing. Saved changes will be written directly to the original location in S3.'
+            )
+
+            const dontShow = localize('AWS.s3.fileViewer.button.dismiss', "Don't show this again")
```

Comment:
Might want to have a separate "OK" button. My thought is that the X is less prominent and this is a big/scary message.

Reply:
I already mentioned having a 'Close' button. The concern was that the notification could look too dense with 3 buttons. 'OK' would be better though since it's smaller.

Reply:
So have 3 buttons?

Reply:
Maybe try it out and see how it looks. If it looks bad then go w/ 2 buttons.

Reply:
It doesn't look bad, I will keep the 3 buttons